base/exports.jl

```diff
@@ -23,6 +23,7 @@ export
     Docs,
     Markdown,
     Threads,
+    Iterators,
```

Comment:
Good point. I tried both ways, and exporting it didn't change the situation much. The way `using` works, if you do `using Iterators` first it loads the package with no warnings. All the other public submodules of Base are exported, so I decided to be consistent. However I'm willing to remove all or most of these exports as part of #5155; some will become top-level modules instead of submodules, and for the remainder we can change the policy on exporting them.